server/src/main/java/org/apache/druid/initialization/ServerInjectorBuilder.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.initialization;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import com.google.inject.Injector;
+import com.google.inject.Module;
+import com.google.inject.TypeLiteral;
+import com.google.inject.multibindings.MapBinder;
+import com.google.inject.multibindings.Multibinder;
+import org.apache.druid.discovery.DruidService;
+import org.apache.druid.discovery.NodeRole;
+import org.apache.druid.guice.annotations.Self;
+
+import java.util.Set;
+
+/**
+ * Initialize Guice for a server. Clients and tests should use
+ * the individual builders to create a non-server environment.
+ */
+public class ServerInjectorBuilder
+{
+  private Injector baseInjector;
+  private Set<NodeRole> nodeRoles;
+  private Iterable<? extends Module> modules;
+
+  /**
+   * Create a server injector. Located here for testing. Should only be
+   * used by {@code GuiceRunnable} (and tests).
+   *
+   * @param nodeRoles the roles which this server provides
+   * @param baseInjector the startup injector
+   * @param modules modules for this server
+   * @return the injector for the server
+   */
+  // TODO: Move to the CLI package, perhaps GuiceRunnable, once
```

Comment:
Please don't commit TODOs. (I know there are some in the codebase, but we generally try to avoid them.) Either do it now, or rewrite it as a regular comment, or raise a follow-up github issue, or simply delete it if it's not a big deal and not going to be done any time soon.